src/main/java/hudson/plugins/accurev/delegates/AbstractModeDelegate.java

```diff
@@ -350,4 +394,90 @@ public void buildEnvVars(AbstractBuild<?, ?> build, Map<String, String> env) {
     protected void buildEnvVarsCustom(AbstractBuild<?, ?> build, Map<String, String> env) {
         // override to put implementation specific values
     }
+    /**
+     * get color type parameter from the AccuRev Version If version less than 6 or equal to 6.0.x the color parameter will be style if
+     * version greater than 6 like 6.1.x or 7 then color parameter will streamStyle
+     * 
+     * @return
+     */
+    private String getStreamTypeParameter() {
+        String fullVersion = GetAccuRevVersion.getAccuRevVersion().trim();
+        String partialversion = fullVersion.substring(fullVersion.indexOf(" ") + 1);
+        String version = partialversion.substring(0, partialversion.indexOf(" "));
+        String[] versionSplits = version.split("\\.");
+        String type = ((Integer.parseInt(versionSplits[0]) < 6) || (Integer.parseInt(versionSplits[0]) == 6 && Integer
+                .parseInt(versionSplits[1]) < 1)) ? "style" : "streamStyle";
+        logger.info("Current AccuRev version " + fullVersion + " color type parameter " + type);
+        return type;
+    }
+
+    /**
+     * Get last transaction build from the jenkins for the currently running project
+     * @return
+     * @throws IOException
+     */
+    private String getLastBuildTransaction() throws IOException {
+        StringBuilder path = new StringBuilder(accurevEnv.get(JENKINS_HOME)).append("\\").append(JOBS).append("\\")
```

Comment:
Their are so many better ways to get the jobs folder location on master.

You can pass the correct object to getLastBuildTransaction

I suggest using the project rootDir.
if the class is Run this is how you get to project
`build.getParent().getRootDir();`
if the class is AbstractProject
`project.getRootDir();`

see
https://github.com/jenkinsci/accurev-plugin/blob/master/src/main/java/hudson/plugins/accurev/AccurevPromoteTrigger.java#L75
https://github.com/jenkinsci/accurev-plugin/blob/master/src/main/java/hudson/plugins/accurev/AccurevPromoteTrigger.java#L67

I would suggest moving that to a utility class perhaps so they both use the same method and file name 👍